php_phongo.c

```diff
@@ -1878,17 +1264,130 @@ static bool php_phongo_apply_wc_options_to_uri(mongoc_uri_t *uri, bson_t *option
 	return true;
 } /* }}} */
 
-static mongoc_client_t *php_phongo_make_mongo_client(php_phongo_manager_t *manager, const mongoc_uri_t *uri, zval *driverOptions TSRMLS_DC) /* {{{ */
+static bool php_phongo_apply_ssl_opts(mongoc_client_t *client, zval *zdriverOptions TSRMLS_DC)
+{
+	mongoc_ssl_opt_t ssl_opt = {0};
+	zval *zoptions;
+	zend_bool free_pem_file = 0, free_pem_pwd = 0, free_ca_file = 0, free_ca_dir = 0, free_crl_file = 0;
+	int plen;
+	bool valid_options;
+
+	if (!zdriverOptions) {
+		return false;
+	}
+
+	zoptions = zdriverOptions;
+
+	/* Use SSL context options if provided for backwards compatibility */
+	if (php_array_existsc(zdriverOptions, "context")) {
+		php_stream_context *context;
+		zval *zcontext;
+
+		zcontext = php_array_fetchc(zdriverOptions, "context");
+
+		context = php_stream_context_from_zval(zcontext, 1);
+
+		if (!context) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "\"context\" driver option is not a valid Stream-Context resource");
+			return false;
+		}
+
+#if PHP_VERSION_ID >= 70000
+		zoptions = php_array_fetchc_array(&context->options, "ssl");
+#else
+		zoptions = php_array_fetchc_array(context->options, "ssl");
+#endif
+
+		if (!zoptions) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Stream-Context resource does not contain \"ssl\" options array");
+			return false;
+		}
+	}
+
+	/* Check canonical option names first and fall back to SSL context options
```

Comment:
I did not bother using both options (canonical and alias) in a single test. All of the tests are just split for using a `$driverOptions` array with canonical options and a stream context with the BC aliases.